### PR TITLE
[PLATFORM-1568]: Add support for log error tracking in datadog 

### PR DESCRIPTION
**Pull request by MaeIsBad**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-1568

### CODE REVIEW

File: tests/e2e/mod.rs [hunk 1]
```diff
@@ -41,7 +37,7 @@ async fn get_spans(f: impl FnOnce()) -> Option<Vec<Span>> {
 
 #[cfg(feature = "traces")]
 #[tokio::test(flavor = "multi_thread")]
-async fn it_sends_traces_to_jaeger() {
+async fn traces_are_sent_to_datadog() {
     let log_message = "hello it_sends_traces_to_jaeger";
```
Comment (cpiemontese) [inline tests/e2e/mod.rs:41]: This should be updated too probably...?

File: tests/e2e/mod.rs
Comment (cpiemontese) [inline tests/e2e/mod.rs:1]: Are there edge-cases or possible overrides happening that we may want to test...?

Comment (MaeIsBad) [reply tests/e2e/mod.rs:1]: Testing logs is a bit more annoying/difficult, and after trying for a while I decided it was too much work to implement right now

File: tests/e2e/mod.rs (hunk 1 above)
Comment (MaeIsBad) [reply tests/e2e/mod.rs:41]: Doesn't really matter, but done for consistency

**Review by cpiemontese — APPROVED** [review]
